### PR TITLE
🧪Introduce ads experiment for intersect-resources

### PR DESCRIPTION
**Pull request by calebcordry**
To determine impact of known issue.

cc/ @jeffkaufman

### CODE REVIEW

**Comment by jeffkaufman** [comment]
Thanks for starting the experiment!

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -53,6 +57,13 @@ const FOCUS_HISTORY_TIMEOUT_ = 1000 * 60; // 1min
 const FOUR_FRAME_DELAY_ = 70;
 const MAX_BUILD_CHUNK_SIZE = 10;
 
+/** @const {!{id: string, control: string, experiment: string}} */
+export const INTERSECT_RESOURCES_EXP = {
```
Comment (dvoytenko) [inline src/service/resources-impl.js:61]: Any chance of putting this into a separate file? I've been much more preferring a separate declaration file lately. We can call it `intersec-resources-exp.js` or such.

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -218,8 +229,11 @@ export class ResourcesImpl {
      */
     this.intersectionObserverCallbackFired_ = false;
 
+    this.divertIntersectResources_();
+
     if (
-      isExperimentOn(this.win, 'intersect-resources') &&
+      getExperimentBranch(this.win, INTERSECT_RESOURCES_EXP.id) ===
```
Comment (dvoytenko) [inline src/service/resources-impl.js:235]: One note: we don't know yet by this point whether we can run this experiment even if we select the branch. That will only happen if later we confirm that the API exists. So the numbers will be somewhat dirty. Is this ok? Or should we reverse and first test eligibility before throwing a coin?

File: src/service/resources-impl.js [hunk 3]
```diff
@@ -341,6 +355,28 @@ export class ResourcesImpl {
     this.schedulePass();
   }
 
+  /**
+   * Select exp vs control for intersect-resources.
+   * @private
+   */
+  divertIntersectResources_() {
```
Comment (dvoytenko) [inline src/service/resources-impl.js:362]: Can we move this function into the same experiment file?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: src/service/resources-impl.js (hunk 1 above)
Comment (calebcordry) [reply src/service/resources-impl.js]: done.

File: src/service/resources-impl.js (hunk 3 above)
Comment (calebcordry) [reply src/service/resources-impl.js]: done.

**Review by dvoytenko — APPROVED** [review]
LG from me on structure. Please confirm what we want to do with the eligibility.

File: src/service/resources-impl.js (hunk 2 above)
Comment (jridgewell) [reply src/service/resources-impl.js:228]: Yah, I think this needs to be done lower. Essentially, try to register the InOb. Then see if we're in the control branch, in which case we'd disconnect it and cleanup.

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: src/service/resources-impl.js (hunk 2 above)
Comment (dvoytenko) [reply src/service/resources-impl.js:228]: The eligibility test is this:

```
function isEligible() {
  try {
    new IntersectionObserver(() => {}, {root: document});
    return true;
  } catch (e) {
    return false;
  }
}
```

Comment (calebcordry) [reply src/service/resources-impl.js:228]: Thanks for the check logic. Switched to only pick branch if InOb supported.

**Review by jridgewell — APPROVED** [review]

**Review by dvoytenko — APPROVED** [review]

**Review by lannka — APPROVED** [review]

**Comment by calebcordry** [comment]
@samouri could you please approve, looks like Justin's approval was before bundle-size bot finished and he is now OOO.

**Review by samouri — APPROVED** [review]

**Review by erwinmombay — APPROVED** [review]

**Pull request merged**